lib/vcloud/core/vapp.rb

```diff
@@ -164,6 +164,16 @@ def power_on
         running?
       end
 
+      # Power off vApp
+      #
+      # @return [Boolean] Returns true if the VM is powered off, false otherwise
+      def power_off
+        raise "Cannot power off a missing vApp." unless id
```

Comment:
Bear in mind that this project is a library and should be agnostic of the particular use case of our internal tooling. Raising an exception is a disruptive action - it will halt the operation of the caller unless they explicitly catch the exception - which is the right thing to do in this case, because you've asked it to stop a vApp that can't be stopped. As a user of the library I wouldn't expect it to continue or silence errors for something that isn't possible. But if that's expected for my particular use case, then I can choose to catch the exception and ignore it (NB: named exception classes can be useful for this).